### PR TITLE
cleanup: run all page cleanup-images in threads.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
The images cleaning is very slow, this PR will increase the number of images cleaned. the main reason is that the directory deleting is taking from 7 to  8 min to complete and we have more than 22000 images to cleanup, the implementation as is today will take ~ 122 days to complete, the current PR should decrease it to ~ 4 days . this is max numbers as not all images has repos to cleanup (if 1/3 of images has repos to clean up this will take 40 days in the current implementation and  1.4 day with the current PR).
 
- default page limit is 30. this PR will handle cleaning all page images in threads.
- retry deleting file.
- unit-tests coverage.
- exit the loop if feature flag is disabled.

FIXES: THEEDGE-3456

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: cmd/cleanup/cleanupimages/cleanupimages.go [hunk 1]
```diff
@@ -23,17 +23,19 @@ var ErrImagesCleanUPNotAvailable = errors.New("images cleanup is not available")
...
 // DefaultDataLimit the default data limit to use when collecting data
 var DefaultDataLimit = 30
 
 // DefaultMaxDataPageNumber the default data pages to handle as preventive way to enter an indefinite loop
 var DefaultMaxDataPageNumber = 1000
 
-// DefaultDeleteFoldersAttempts the default delete folder attempts
-var DefaultDeleteFoldersAttempts = 10
+// DefaultDeleteAttempts the default delete folder/files attempts
+var DefaultDeleteAttempts = 10
```
Comment (loadtheaccumulator) [inline cmd/cleanup/cleanupimages/cleanupimages.go:35]: Thoughts on making this configurable as an Env var?

**Review by loadtheaccumulator — COMMENTED** [review]
Looks good. Just the two questions.

File: cmd/cleanup/cleanupimages/cleanupimages.go [hunk 2]
```diff
@@ -23,17 +23,19 @@ var ErrImagesCleanUPNotAvailable = errors.New("images cleanup is not available")
...
 
-// DefaultDeleteFoldersRetryDelay the default delete folder delay before a retry
-var DefaultDeleteFoldersRetryDelay = 5 * time.Second
+// DefaultDeleteRetryDelay the default delete folder/files delay before a retry
+var DefaultDeleteRetryDelay = 5 * time.Second
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply cmd/cleanup/cleanupimages/cleanupimages.go]: done

**Comment by ldjebran** [comment]
@loadtheaccumulator your comments has been addressed please revisit.

**Review by loadtheaccumulator — APPROVED** [review]
lgtm

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**